pkg/performanceprofile/cmd/render/render.go

```diff
@@ -58,14 +65,107 @@ func init() {
 	utilruntime.Must(performancev2.AddToScheme(manifestScheme))
 	utilruntime.Must(apicfgv1.Install(manifestScheme))
 	utilruntime.Must(mcfgv1.Install(manifestScheme))
+	utilruntime.Must(tunev1.AddToScheme(manifestScheme))
 	codecFactory = serializer.NewCodecFactory(manifestScheme)
 	runtimeDecoder = codecFactory.UniversalDecoder(
 		performancev2.GroupVersion,
 		apicfgv1.GroupVersion,
 		mcfgv1.GroupVersion,
+		tunev1.SchemeGroupVersion,
 	)
 }
 
+func expandCpuSet(cpuSet string) string {
+	re := regexp.MustCompile(`(\d+-\d+)`)
+	matches := re.FindAllString(cpuSet, -1)
+
+	var coreRanges []string
+	coreRanges = append(coreRanges, matches...)
+
+	var expandedCpus []string
+	for _, rangeStr := range coreRanges {
+		ranges := strings.Split(rangeStr, "-")
+		start, _ := strconv.Atoi(ranges[0])
+		end, _ := strconv.Atoi(ranges[1])
+
+		for i := start; i <= end; i++ {
+			expandedCpus = append(expandedCpus, strconv.Itoa(i))
+		}
+	}
+
+	return strings.Join(expandedCpus, ",")
+}
+
+func coresToHexMask(coresExp string) string {
```

Comment:
likewise, I think we have existing utilities to cover this usecase

Reply:
Fixed